bin/cron/zendesk_slack_report

```diff
@@ -0,0 +1,145 @@
+MONITORING_GROUPS = {
+  # "slack-room-name" => ["Zendesk group name", "Zendesk group name", ...]
+  "learning-platform" => ["Learning Platform"],
+  # "developers" => ["Developers"],
+  # "plc-engineering" => ["PLC Tools"],
+  # "privacy" => ["Privacy"],
+  # "star-labs-cabal" => ["*Labs"],
+  # "user-accounts" => ["Accounts"],
+}
+
+# Zendesk group -> Zendesk filter id mappings for helpful links (optional)
+# rubocop:disable Style/NumericLiterals
+GROUP_FILTERS = {
+  "Developers" => 47222126,
+  "Learning Platform" => 360082967212,
+  "PLC Tools" => 114103940251,
+  "*Labs" => 301129028,
+}
+# rubocop:enable Style/NumericLiterals
+
+class Zendesk
+  include HTTParty
+  base_uri 'https://codeorg.zendesk.com'
+
+  def groups
+    @groups ||= self.class.
+      get(
+        "/api/v2/groups.json",
+        {basic_auth: {username: 'dev@code.org/token', password: CDO.zendesk_dev_token}}
+      ).
+      parsed_response["groups"].
+      map {|g| [g["id"], g["name"]]}.
+      to_h
+  end
+
+  def search(groups:, statuses:)
+    query = Zendesk.build_query(groups: groups, statuses: statuses)
+    page = 1
+    results = []
+    loop do
+      response = self.class.get(
+        "/api/v2/search.json",
+        {
+          basic_auth: {username: 'dev@code.org/token', password: CDO.zendesk_dev_token},
+          query: {page: page, query: query}
+        }
+      ).parsed_response
+      results += response["results"]
+      if response["next_page"]
+        page += 1
+      else
+        break
+      end
+    end
+    results
+  end
+
+  def self.build_query(groups: [], statuses: [])
+    [
+      statuses.map {|s| "status:#{s}"},
+      groups.map {|g| "group:\"#{g}\""}
+    ].flatten.join(' ')
+  end
+
+  def self.group_url(group:, statuses:)
+    filter_id = GROUP_FILTERS[group]
+    if filter_id
+      return "#{base_uri}/agent/filters/#{filter_id}"
+    end
+
+    query = build_query(groups: [group], statuses: statuses)
+    # Zendesk not friendly to '+' for spaces?
+    querystring = URI.encode_www_form(q: query).gsub('+', '%20')
+    "#{base_uri}/agent/search/1?#{querystring}"
+  end
+end
+
+#
+# MAIN
+#
+#
+
+# Get tickets from Zendesk API
+zendesk = Zendesk.new
+results = zendesk.search(
+  statuses: STATUSES_TO_REPORT,
+  groups: MONITORING_GROUPS.values.flatten
+)
+
+# Ticket count by group name
+counts = results.
+  group_by {|x| x["group_id"]}.
+  map {|group_id, tickets| [zendesk.groups[group_id], tickets.count]}.
+  to_h
+
+# Report out for each Slack room
+MONITORING_GROUPS.each do |slack_room, groups|
+  group_messages = groups.map do |group_name|
+    count = counts[group_name] || 0
+    next nil if count == 0
+
+    url = Zendesk.group_url(group: group_name, statuses: STATUSES_TO_REPORT)
+    statuses = STATUSES_TO_REPORT.join('/')
+    "<#{url}|#{group_name} has #{count} #{statuses} Zendesk tickets.>"
+  end.compact
+
+  # Don't post if there's nothing to report
+  next if group_messages.empty?
```

Comment:
Note, we intentionally post _nothing_ if the count is zero for a channel.

I'd like to keep Slack notifications actionable, and zero tickets is not an actionable notification.  If we want this for metrics purposes, let's extend this script to log counts to redshift or something.